.github/workflows/dogfood-automated-policy-updates.yml

```
@@ -0,0 +1,28 @@
+name: "Automated policy updates for dogfood"
+
+on:
+  schedule:
+    - cron: '0 */6 * * *'  # Run every 6 hours
+  workflow_dispatch:     # Allow manual trigger
+
+jobs:
+  update-macos-versions:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Git
+        run: |
+          git config --global user.name "GitHub Action"
+          git config --global user.email "action@github.com"
+
+      - name: Run macOS version update script
+        run: |
+          chmod +x ./.github/scripts/dogfood-policy-updater-latest-macos.sh
+          ./.github/scripts/dogfood-policy-updater-latest-macos.sh
+        env:
+          DOGFOOD_AUTOMATION_TOKEN: ${{ secrets.DOGFOOD_AUTOMATION_TOKEN }}
```

```suggestion
          DOGFOOD_AUTOMATION_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

I was mistaken. I realized you can just pass in the `GITHUB_TOKEN` secret to your script instead of the unique environment variable. Once this merged I'll delete the old `DOGFOOD_AUTOMATION_TOKEN`.

---

@lukeheath Thanks for the pointer on `GITHUB_TOKEN`. I spent quite a bit of time trying to figure out how to use this without success. There are definitely use cases that I'll incorporate `GITHUB_TOKEN` into going forward but in this specific case, `GITHUB_TOKEN` cannot be used to create pull requests. Only "GitHub App user access tokens", "GitHub App installation access tokens", and "Personal access tokens" can be used with the create pull request endpoint. Since we don't want to commit to main without review, it might be best to keep the env variable tied to a personal access token. Thoughts?

---

That makes sense. Everywhere else we create PRs uses the `peter-evans/create-pull-request` action [like we do here](https://github.com/fleetdm/fleet/blob/main/.github/workflows/update-osquery-versions.yml). In that example, we call out to a script to determine if there is a new osquery version, and if so, to create a commit.  the `create-pull-request` action always runs, but only creates a pull request if there is a new commit, and it has access to the same `git` repo, so it will see the commit from the script. 

The benefit is not having to use an associated PAT, but honestly we're using PATs in other workflows so it's not a big deal. Totally not a blocker, just a thought for the future.